libs/sql-schema-describer/src/postgres.rs

```diff
@@ -57,6 +57,33 @@ impl<'a> super::SqlSchemaDescriberBackend for SqlSchemaDescriber<'a> {
         let table_names = self.get_table_names(schema).await?;
         let mut tables = Vec::with_capacity(table_names.len());
 
+        if self.is_cockroach() {
+            // Currently, we ignore all hidden columns from CockroachDB.
+            // However, these still show up from get_indices, where CockroachDB can place implicit
+            // columns in front of indexes and primary keys.
+            // For now, remove all hidden columns from the indexes and PK, removing them as an index
+            // or PK if every column is implicit.
+            for (table_name, table_indexes) in indexes.iter_mut() {
+                let mut table_columns = HashSet::new();
+                if let Some(val) = columns.get(table_name) {
+                    for c in val {
+                        table_columns.insert(c.name.to_string());
+                    }
+                }
+                let (indexes, table_pk_wrapped) = table_indexes;
+
+                let table_pk = table_pk_wrapped.as_mut().unwrap();
+                table_pk.columns.retain(|c| table_columns.contains(c));
+                if table_pk.columns.len() == 0 {
+                    table_indexes.1 = None;
+                }
+                for index in indexes.iter_mut() {
+                    index.columns.retain(|c| table_columns.contains(c))
+                }
+                indexes.retain(|i| i.columns.len() > 0);
+            }
```

Comment:
That can definitely be made a lot shorter with iterators, but let's do this as a cleanup PR later, I think this should work.